src/main.ts

```diff
@@ -1,16 +1,21 @@
 import * as core from '@actions/core'
-import {wait} from './wait'
+import {generateChangelog} from './generate-changelog'
 
 async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
-    core.debug(`Waiting ${ms} milliseconds ...`) // debug is only output if you set the secret `ACTIONS_STEP_DEBUG` to true
+    const github_auth_token: string = core.getInput('github_auth_token', {
+      required: true
+    })
+    const current_version: string = core.getInput('version', {required: true})
+    const channel: string = core.getInput('channel', {required: true})
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    const changelog = await generateChangelog(
```

Comment:
ooc: if it fails (for example, won't be able to correctly identify the previous release) what will happen? will we just fail to create a stable release?

Reply:
I believe (@alokedesai correct me if I'm wrong here) that will cause this individual step to fail. So it would be up to the underlying Action we create to make sure this is called in a way that doesn't fail the whole build if it fails.

Reply:
that's correct--we'll run this in our post_release section of the release workflow as the last step, so the entire release will still run correctly regardless of whether this action fails: https://github.com/warpdotdev/warp-internal/blob/master/.github/workflows/stable_release.yml#L110.

Additionally, i'll set the `continue_on_error` bit on the workflow when calling it to avoid unnecessary failures for the entire workflow if we ever add more steps to the `post_release` job.